packages/earn-controller/src/EarnController.ts

```diff
@@ -500,6 +495,25 @@ export class EarnController extends BaseController<
     return convertHexToDecimal(chainId);
   }
 
+  /**
+   * Ensures chainId is compatible with pooled-staking. Falls back to Ethereum Mainnet if chainId is not supported.
+   *
+   * @returns The current chain id in decimal. Ethereum Mainnet if it's not an ethereum chain.
+   */
+  #getActivePooledStakingChainId(): number {
+    const activeChainId = this.#getCurrentChainId();
+
+    if (
+      !activeChainId ||
+      !this.#supportedPooledStakingChains.includes(toHex(activeChainId))
+    ) {
+      // Fallback to Ethereum Mainnet if chainId is not supported.
+      return convertHexToDecimal(CHAIN_IDS.MAINNET);
+    }
+
+    return activeChainId;
+  }
```

Comment:
I'm still getting to grips with what we need to do on Earn side regarding GNS removal, but it looks like if we are removing the `chainId` params of the pooled staking calls in favour of using this function within them, then we are increasing our coupling to `getCurrentChainId()` and therefore `NetworkController:getState` which is something we want to avoid? 

If instead we want to always pass in a `chainId` to the pooled staking calls then we can still use this logic to check if it's a supported chain but without calling `NetworkController:getState`. Let me know if I'm getting this right and if this is an interim fix to get it working without having to change the mobile code also!

Reply:
All good, I think this as a stopgap is fine as it is so I'll approve to unblock you!